CONTRIBUTING.md

```diff
@@ -491,19 +491,27 @@ To get a sense for what changes are considered mass rebuilds, see [previously me
 ## Commit conventions
 [commit-conventions]: #commit-conventions
 
-- Create a commit for each logical unit.
+- Commits should be independent and understandable by themselves.
 
-- Check for unnecessary whitespace with `git diff --check` before committing.
+  For example:
+  - For example, when adding yourself as maintainer in the same pull request,
+    make a separate commit with the message `maintainers: add <handle>`.
+    Add the commit before those making changes to the package or module.
+    See [Nixpkgs Maintainers](./maintainers/README.md) for details.
 
-- If you have commits `pkg-name: oh, forgot to insert whitespace`: squash commits in this case. Use `git rebase -i`.
-  See [Squashing Commits](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History#_squashing) for additional information.
+- Later commits that undo changes from earlier commits should be combined.
 
-- For consistency, there should not be a period at the end of the commit message's summary line (the first line of the commit message).
+  For example:
+  - If you have commits `pkg-name: oh, forgot to insert whitespace`: squash commits in this case. Use `git rebase -i`.
+    See [Squashing Commits](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History#_squashing) for additional information.
+
+> [!Note]
```

Comment:
:+1: I guess we will eventually want rule like this to apply to all guidelines as per https://github.com/NixOS/nixpkgs/issues/387072 but I guess limiting the scope for now is a good start.